wicket-core/src/main/java/org/apache/wicket/protocol/http/request/WebClientInfo.java

```diff
@@ -133,24 +140,99 @@ public final String getUserAgent()
 	}
 
 	/**
-	 * returns the user agent string (lower case).
+	 * Initializes the {@link WebClientInfo} user agent detection. This can be overridden to choose
+	 * a different detection as YAUAA (https://github.com/nielsbasjes/yauaa) - if you do so, you
+	 * might exclude the maven dependency from your project in favor of a different framework.
+	 */
```

Comment:
No, you can't exclude the dependency - with UserAgentAnalyzer in the signature of #detectBrowserProperties(UserAgentAnalyzer) you will get a NoClassDefFoundError if the dependency is not present.

Reply:
Ok, then keep the dependency and just override it. We have to adjust the documentation here.